app/controllers/devise/registrations_controller.rb

```diff
@@ -83,9 +83,10 @@ def cancel
   protected
 
   def update_needs_confirmation?(resource, previous)
-    resource.respond_to?(:pending_reconfirmation?) &&
-      resource.pending_reconfirmation? &&
-      previous != resource.unconfirmed_email
+    resource.respond_to?(:can_send_reconfirmation?) &&
```

Comment:
which flash notice message? I don't think I follow.

I'm trying to go down the path that this patch should only change the mailer template we use for the reconfirmation cases, and shouldn't affect anything on the user flow or UI feedback we already have - that's the whole scope of what was discussed on plataformatec/devise#4145.